pandas/core/indexes/datetimelike.py

```diff
@@ -664,13 +665,17 @@ def maybe_unwrap_index(obj):
     Parameters
     ----------
     obj : object
+    unwrap_series : bool
+        Whether to also unwrap Series objects.
 
     Returns
     -------
     unwrapped object
     """
     if isinstance(obj, ABCIndexClass):
         return obj._data
+    elif isinstance(obj, ABCSeries) and unwrap_series:
+        obj = obj._values
```

Comment:
is the preferred usage for this now `obj.array`?

Reply:
not a big fan of having a function that operates with an arg like this, maybe make a ``maybe_unwrap_series`` instead?

Reply:
I'd be OK with renaming this function to not be index-specific, but not wild about adding another function that we're just gonna use in one place.

Reply:
if its only 1 place, then just inline it , no?

Reply:
That works, but I find this clearer since we're already calling this function in that location.  It makes it clear that doing it in location A but not location B is intentional.

Reply:
this arg makes it very ugly, ok w/o it.